test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -638,9 +638,12 @@ TEST_F(ClientLbEnd2endTest, ChannelStateConnectingWhenResolving) {
   EXPECT_EQ(channel->GetState(true /* try_to_connect */), GRPC_CHANNEL_IDLE);
   // Now that the channel is trying to connect, we should get to state
   // CONNECTING.
-  while (channel->GetState(false) == GRPC_CHANNEL_IDLE) {
-    std::this_thread::sleep_for(std::chrono::milliseconds(10));
-  }
+  ASSERT_TRUE(
+      WaitForChannelState(channel.get(), [&](grpc_connectivity_state state) {
+        if (state == GRPC_CHANNEL_IDLE) return false;
+        EXPECT_EQ(state, GRPC_CHANNEL_CONNECTING);
+        return true;
+      }));
   EXPECT_EQ(channel->GetState(false /* try_to_connect */),
```

Comment:
This expectation is no longer needed with the change you made immediately above.

Reply:
done